corehq/apps/enterprise/tests/test_tasks.py

```diff
@@ -0,0 +1,140 @@
+        cls.domains1 = [
+            create_domain('domain-account1-001'),
+            create_domain('domain-account1-002'),
+        ]
+        add_domains_to_enterprise_account(
+            cls.billing_account1,
+            cls.domains1,
+            enterprise_plan,
+            one_year_ago
+        )
+        cls.emw_settings_active = EnterpriseMobileWorkerSettings.objects.create(
+            account=cls.billing_account1,
+            enable_auto_deactivation=True,
+            allow_custom_deactivation=True,
+        )
+
+        cls.billing_account2 = get_enterprise_account()
+        cls.domains2 = [
+            create_domain('domain-account2-001'),
+            create_domain('domain-account2-002'),
+            create_domain('domain-account2-003'),
+        ]
+        add_domains_to_enterprise_account(
+            cls.billing_account2,
+            cls.domains2,
+            enterprise_plan,
+            one_year_ago
+        )
+        cls.emw_settings_no_custom = EnterpriseMobileWorkerSettings.objects.create(
+            account=cls.billing_account2,
+            enable_auto_deactivation=True,
+        )
+
+        cls.billing_account3 = get_enterprise_account()
+        cls.domains3 = [
+            create_domain('domain-account3-001'),
+            create_domain('domain-account3-002'),
+        ]
+        add_domains_to_enterprise_account(
+            cls.billing_account3,
+            cls.domains3,
+            enterprise_plan,
+            one_year_ago
+        )
+        cls.emw_settings_inactive = EnterpriseMobileWorkerSettings.objects.create(
+            account=cls.billing_account3,
+            enable_auto_deactivation=False,
+            allow_custom_deactivation=False,
+        )
+
+        cls.billing_account4 = get_enterprise_account()
+        cls.domains4 = [
+            create_domain('domain-account4-001'),
+        ]
+        add_domains_to_enterprise_account(
+            cls.billing_account4,
+            cls.domains4,
+            enterprise_plan,
+            one_year_ago
+        )
+        cls.emw_settings_inactive = EnterpriseMobileWorkerSettings.objects.create(
+            account=cls.billing_account4,
+            enable_auto_deactivation=False,
+            allow_custom_deactivation=True,
+        )
+
+    @classmethod
+    def tearDownClass(cls):
+        EnterpriseMobileWorkerSettings.objects.all().delete()
+        for domains in [cls.domains1, cls.domains2, cls.domains3, cls.domains4]:
+            for domain in domains:
+                domain.delete()
+        super().tearDownClass()
+
+    @mock.patch('corehq.apps.enterprise.tasks.metrics_gauge')
+    @mock.patch('corehq.apps.enterprise.tasks.EnterpriseMobileWorkerSettings'
+                '.deactivate_mobile_workers_by_inactivity')
+    @mock.patch('corehq.apps.enterprise.tasks.DeactivateMobileWorkerTrigger'
+                '.deactivate_mobile_workers')
+    def test_auto_deactivate_mobile_workers_task(
```

Comment:
FYI this failed in a seemingly unrelated PR (https://github.com/dimagi/commcare-hq/pull/31178), it might be flaky.
```
2022-03-01T16:37:05.3999975Z ======================================================================
2022-03-01T16:37:05.4004109Z FAIL: corehq.apps.enterprise.tests.test_tasks:TestAutoDeactivateMobileWorkersTask.test_auto_deactivate_mobile_workers_task
2022-03-01T16:37:05.4006912Z ----------------------------------------------------------------------
2022-03-01T16:37:05.4014696Z Traceback (most recent call last):
2022-03-01T16:37:05.4015112Z   File "/usr/local/lib/python3.9/unittest/mock.py", line 1336, in patched
2022-03-01T16:37:05.4015509Z     return func(*newargs, **newkeywargs)
2022-03-01T16:37:05.4016398Z   File "/mnt/commcare-hq-ro/corehq/apps/enterprise/tests/test_tasks.py", line 117, in test_auto_deactivate_mobile_workers_task
2022-03-01T16:37:05.4017045Z     self.assertListEqual(
2022-03-01T16:37:05.4019265Z AssertionError: Lists differ: ['domain-account1-001', 'domain-account1-002', 'domain-accoun[49 chars]003'] != ['domain-account1-002', 'domain-account1-001', 'domain-accoun[49 chars]001']
2022-03-01T16:37:05.4019710Z 
2022-03-01T16:37:05.4019952Z First differing element 0:
2022-03-01T16:37:05.4021286Z 'domain-account1-001'
2022-03-01T16:37:05.4021647Z 'domain-account1-002'
2022-03-01T16:37:05.4022740Z 
2022-03-01T16:37:05.4022936Z - ['domain-account1-001',
2022-03-01T16:37:05.4023221Z ?                     ^
2022-03-01T16:37:05.4023471Z 
2022-03-01T16:37:05.4023655Z + ['domain-account1-002',
2022-03-01T16:37:05.4023917Z ?                     ^
2022-03-01T16:37:05.4024077Z 
2022-03-01T16:37:05.4024262Z -  'domain-account1-002',
2022-03-01T16:37:05.4024535Z ?                     ^
2022-03-01T16:37:05.4024849Z 
2022-03-01T16:37:05.4025034Z +  'domain-account1-001',
2022-03-01T16:37:05.4025311Z ?                     ^
2022-03-01T16:37:05.4025470Z 
2022-03-01T16:37:05.4025659Z -  'domain-account2-001',
2022-03-01T16:37:05.4025936Z ?                     ^
2022-03-01T16:37:05.4026096Z 
2022-03-01T16:37:05.4026366Z +  'domain-account2-003',
2022-03-01T16:37:05.4026640Z ?                     ^
2022-03-01T16:37:05.4026797Z 
2022-03-01T16:37:05.4026984Z    'domain-account2-002',
2022-03-01T16:37:05.4027877Z -  'domain-account2-003']
2022-03-01T16:37:05.4028168Z ?                     ^
2022-03-01T16:37:05.4028330Z 
2022-03-01T16:37:05.4028538Z +  'domain-account2-001']
2022-03-01T16:37:05.4028817Z ?                     ^
2022-03-01T16:37:05.4028976Z 
```

Reply:
@orangejenny https://github.com/dimagi/commcare-hq/pull/31185